### PR TITLE
feat(arm): add CKV_AZURE_173 to ensure that API management uses at least TLS 1.2

### PR DESCRIPTION
**Pull request by goldyBH**
APIManagementMinTLS12
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
  

    

## Description

ensure that API management uses at least TLS 1.2

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/APIManagementMinTLS12.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def __init__(self) -> None:
+        name = "Ensure API management uses at least TLS 1.2"
+        id = "CKV_AZURE_173"
+        supported_resources = ("Microsoft.ApiManagement/service",)
+        categories = (CheckCategories.ENCRYPTION,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        properties = conf.get('properties', {})
+        if 'security' in properties:
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/APIManagementMinTLS12.py:17]: There is no field `security` in `Microsoft.ApiManagement/service` type

**Comment by goldyBH** [comment]
What do I need to do to fix the security?

File: checkov/arm/checks/resource/APIManagementMinTLS12.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        id = "CKV_AZURE_173"
+        supported_resources = ("Microsoft.ApiManagement/service",)
+        categories = (CheckCategories.ENCRYPTION,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        properties = conf.get('properties', {})
+        if 'identity' in properties:
+            identity = properties['identity']
+            if 'enableFrontendTls10' in identity:
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/APIManagementMinTLS12.py:19]: this is the identity structure - 
"identity": {
    "type": "string",
    "userAssignedIdentities": {
      "{customized property}": {
        "clientId": "string",
        "principalId": "string"
      }
    }
  }

**Comment by ChanochShayner** [comment]
Implemented here - https://github.com/bridgecrewio/checkov/pull/6478
Closing.

**Pull request closed without merging**